util/src/main/java/org/folio/reservoir/util/readstream/XmlFixerMapper.java

```diff
@@ -34,18 +57,44 @@ public void push(Buffer buffer) {
     tail = 0;
     for (front = 0; front < input.length(); front++) {
       byte leadingByte = input.getByte(front);
-      if (leadingByte > 0 && leadingByte < 32
-          && leadingByte != 9 && leadingByte != 10 && leadingByte != 13) {
-        result.appendBuffer(input, tail, front - tail);
-        addFix();
-      } else if (leadingByte == '&' && handleEntity(input)) {
-        return;
+      if (leadingByte < 0) {
```

Comment:
Could we extract those checks to private static functions that make the code readable? E.g "isSequence".

Reply:
I'd rather not. This would be a call for every byte. No. I'm going to make it more readable by using the bit-patterns.

Reply:
@adamdickmeiss JVM will inline small methods like this. You can check to confirm, https://www.baeldung.com/jvm-method-inlining

Reply:
So youi want all & X to be inline functions ???

Reply:
No, this is good. Just saying that there should be no perf penalty. I would just use constants for the code points, these are always inlined and it's sufficient to make it readable.

Reply:
The point I'm also making is that if you make a method you realize that if it's going to be sef-sufficient and do a job you need to be careful. Example:
```
  if (c < 32)   // or if (c < ' ')
```
is valid in this context. Eg that we know it has bit 7 = 0 (&128). So if you make a method isControl in order for it to be useful you need to use the contents if (c >= 0 && < c 32) .. But that's a redundant condition. It matters here. You can also call the function checkForControlIfBit7is0 ,.. oh well.

Reply:
Yes, point taken. Then I would just replace it with `if  (c < VALID_CHARS_START)` which immediately tells you what the point is. Same for for `if ((c & 1_BYTE_SEQ_START) == 0)`.

Reply:
I've made some methods now .. Changed the checks so they don't depend on "anything" else. That's smart standard (UTF-8 that is).